README.md

```diff
@@ -1,2 +1,140 @@
+`aspect` field must correspond to an endpoint defined as a part of REST API.
+More details below.
+* `friendly_name` - user-readable name, that will be displayed to user
+* `query_format` - defines, what parameters should be defined to obtain summary
+of the analysis. It should follow the format of JSON-schema
+* `output_type` - one of the following: `table` or `plot`
+
+Output types are described further [below](#output-types).
+
+## Result Access Endpoint
+
+Endpoint `/results/<task-name>/<aspect>` should handle `POST` requests with
+query format defined as above. Response should return code `200` for existing
+result and `404` for nonexistent one. Expected format of the response is
+dependent on the specification already provided. For more details check
+[output types](#output-types).
+
+## Computational Task
+
+Computational tasks should be defined as a part of worker API, in a form
+of [Celery task](http://docs.celeryproject.org/en/latest/userguide/tasks.html).
+The task should accept input defined exactly as declared in previous point.
+
+According to the [documentation of Celery](http://docs.celeryproject.org/en/latest/userguide/calling.html#serializers)
+all the parameters of scheduled task are serialized and then exchanged
+between this node and workers. To limit the overhead, input should be small,
+i.e. no dataset should be sent as an argument. Instead, a dataset name can be
+used, and dataset should be loaded from the disk locally.
+
+Result of finished task should be saved to disk. We'll put more details soon.
+
+## Discovery
+
+Each worker will be found at `analysis-api:2004/api/workers?refresh=True`,
+with worker name before the `@` sign and available task names in `registered`
+field. To make worker discoverable, make sure worker name is the same as its
+hostname. Task names should be constructed like: `category_name.analysis_name`
+(e.g. `analysis.divik`, `modeling.gmm`).
+
+Considered categories are, consecutively:
+
+* preprocessing
+* modeling
+* analysis
+
+# Output types
+
+## `table`
+
+This type of output will be interpreted as a table with multiple columns, for
+example such JSON:
+
+```
+{
+    "columns": [
+        {
+            "key": "x",
+            "name": "Some variable",
+        },
+        {
+            "key": "y",
+            "name": "Another variable",
+        },
+        {
+            "key": "z",
+            "name": "Some other variable",
+        },
+    ],
+    "data": [
+        {"x": 1, "y": 2, "z": 3},
+        {"x": 4, "y": 5, "z": 6},
+        ...
+    ]
+}
+```
+
+will be transformed into table like this:
+
+| Some variable | Another variable | Some other variable |
+| ------------- | ---------------- | ------------------- |
+| 1             | 2                | 3                   |
+| 4             | 5                | 6                   |
+| ...           | ...              | ...                 |
+
+## `plot`
+
```

Comment:
Only plot? What with spectrum, heatmap, boxplots?

Reply:
This is any plot supported by Plotly, until we have our wrappers ready. Having heatmap, boxplot, etc. in scalable manner, we will be able to migrate onto our stable interface for this. Now let's focus on integration of these components.